tests/gitOpsApiTests.js

```diff
@@ -104,36 +115,52 @@ jqUnit.test("Success cases for all API functions - ", function () {
 			}],
 			commitMessage: "A test commit from commitMultipleFiles()"
 		});
-	}).then((res) => {
-		jqUnit.assertTrue("commitMultipleFiles() completes successfully.", res.includes("successfully."));
-		return gitOpsApi.issuePullRequest(octokit, {
+		jqUnit.assertTrue("commitMultipleFiles() completes successfully.", response.includes("successfully."));
+
+		response = await gitOpsApi.issuePullRequest(octokit, {
 			repoOwner: repoOwner,
 			repoName: repoName,
 			issuerGithubId: repoOwner,
 			branchName: branchName,
 			pullRequestTitle: "A test pull request issued by issuePullRequest()"
 		});
-	}).then(function (res) {
-		jqUnit.assertTrue("The pull request url is returned", res.startsWith("https://github.com/"));
-		return gitOpsApi.deleteBranch(octokit, {
+		jqUnit.assertTrue("The pull request url is returned", response.startsWith("https://github.com/"));
+
+		response = await gitOpsApi.deleteBranch(octokit, {
 			repoOwner: repoOwner,
 			repoName: repoName,
 			branchName: branchName
 		});
-	}).then(function (res) {
-		jqUnit.assertTrue("deleteBranch() completes successfully.", res.includes("has been deleted successfully"));
-	}).catch(function (error) {
-		gitOpsApi.deleteBranch(octokit, {
-			repoOwner: repoOwner,
-			repoName: repoName,
-			branchName: branchName
-		}).then(() => {
+		jqUnit.assertTrue("deleteBranch() completes successfully.", response.includes("has been deleted successfully"));
+	} catch (error) {
+		try {
+			response = await gitOpsApi.deleteBranch(octokit, {
```

Comment:
Note that in the latest JQUnit, these kinds of async tests can be simplified a bit further - by writing it as jqUnit.test rather than jqUnit.asyncTest and then returning the last promise used in the function (that is, this one), you can leave out the final jqUnit.start()

Reply:
Ah, it looks like you have already done this for all tests other than this one